src/logger/Logger.cpp

```diff
@@ -0,0 +1,194 @@
+#include "kompute/logger/Logger.hpp"
+
+#include <cassert>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <mutex>
+#include <spdlog/async.h>
+#include <spdlog/common.h>
+#include <spdlog/logger.h>
+#include <spdlog/sinks/rotating_file_sink.h>
+#include <spdlog/sinks/stdout_color_sinks.h>
+#include <spdlog/spdlog.h>
+
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#ifdef _WIN32
+#include <iso646.h>
+#include <direct.h>
+#endif // _WIN32
+
+namespace logger {
+constexpr int THREAD_QUEUE_LENGTH = 8192;
+constexpr int FILE_ROTATION_TIME = 1048576 * 5;
+
+/**
+ * Should be replaced with:
+ * std::filesystem::exists(path)
+ * when switching to cpp17
+ **/
+bool
+exists(const std::string& path)
+{
+    struct stat info
+    {};
+    if (stat(path.c_str(), &info) != 0) {
+        // std::cerr << "Failed to check if '" << path
+        //           << "' exists. Cannot access!\n";
+        // assert(false);
+        return false;
+    }
+    return info.st_mode & S_IFDIR;
+}
+
+/**
+ * Based on: https://stackoverflow.com/a/35109823
+ * Should be replaced with:
+ * std::filesystem::create_directory(path);
+ * when switching to cpp17
+ **/
+void
+createDir(const std::string& path)
```

Comment:
What is the reason for having all these helpers? It seems quite opinionated on logic that would be application specific on how logger files should be handled (perhaps could be relevant in the tests/ but not sure in the main module)

Reply:
They allow storing logs inside a directory and performing proper log rotation. Just printing logs on the screen is in my eyes not enough. They should be made persistent somehow.